tests/e2e/e2e.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+# Copyright 2019 Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eux
+
+export ARTIFACTS="${ARTIFACTS:-$(mktemp -d)}"
+
+function cleanup_kind_cluster() {
+    kind export logs --name istio-testing "${ARTIFACTS}/kind"
+#    kind delete cluster --name=istio-testing
```

Comment:
This is a must have! see https://github.com/kubernetes-sigs/kind/issues/759. We should have a consolidated kind library. Maybe in common files? Opened https://github.com/istio/istio/issues/17948 to track this

Reply:
I agree a consolidated setup library would be great.  When I come up for air after 1.4, I had planned to work a bit more on cross-repo gating and make things generic enough so that different components can use the same set of scripts...

Reply:
comment removed from line 23.